presto-spi/src/main/java/com/facebook/presto/spi/session/SessionPropertyConfigurationManager.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.session;
+
+import java.util.List;
+
+public interface SessionPropertyConfigurationManager
+{
+    /**
+     * The list of SessionPropertyRules will be applied in the same order they
+     * appear in the list. Which means, if multiple rules would change the same
+     * session property, the later one will overwrite the values from the earlier ones.
+     */
+    List<? extends SessionPropertyRule> getOrderedRules();
```

Comment:
I'm not sure a list of rules make sense. If the plugin wants to have a list of rules internally, that's an implementation detail. But from the perspective of this API, it's going to give the plugin a context and the plugin will return new session properties.